Models/AutoEncoder3D.py

```diff
@@ -1,87 +1,167 @@
         super().__init__()
         self.n_classes = 1
-        self.backbone = models.resnet50(pretrained=True) 
-        self.model= torch.nn.Sequential(
-            self.unet_model,
-            torch.nn.LazyLinear(128),
-            torch.nn.LazyLinear(self.n_classes)            
-        )
-        summary(self.model.to('cuda'), (2,160,160,40))
+        module_str = config['MODEL']['Backbone']
+        parameters = config['MODEL_PARAMETERS']
+        self.config = config
+
+        model_str = 'nets.' + module_str + '(**parameters)'
+        self.model = eval(model_str)
+        # only use network for features
+
+        summary(self.model.to('cuda'), (config['MODEL']['batch_size'], 1, *config['DATA']['dim']))
         self.accuracy = torchmetrics.AUC(reorder=True)
-        self.loss_fcn = torch.nn.BCEWithLogitsLoss()
+        self.loss_fcn = torch.nn.MSELoss()
 
-    def forward(self, x):
-        return self.model(x)
+    def forward(self, datadict):
+        key = list(datadict.keys())[0]
+        return self.model(datadict[key])
 
-    def training_step(self, batch,batch_idx):
-        image,label = batch
-        prediction  = self.forward(image)
-        loss = self.loss_fcn(prediction.squeeze(), label)
+    def training_step(self, batch, batch_idx):
+        image_dict, label = batch
+        prediction = self.forward(image_dict)
+        key = list(image_dict.keys())[0]
+        loss = self.loss_fcn(prediction, image_dict[key])
         self.log("loss", loss)
-        return {"loss":loss,"prediction":prediction.squeeze(),"label":label}
-
-    def validation_step(self, batch,batch_idx):
-        image,label = batch
-        prediction  = self.forward(image)
-        loss = self.loss_fcn(prediction.squeeze(), label)
-        return {"loss":loss,"prediction":prediction.squeeze(),"label":label}
-        
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        image_dict, label = batch
+        prediction = self.forward(image_dict)
+        key = list(image_dict.keys())[0]
+        loss = self.loss_fcn(prediction, image_dict[key])
+
+        if batch_idx < 3:
+            im_st = image_dict[key][0, 0, 5, :, :]
+            plt.imshow(im_st.cpu())
+            plt.title('standard image')
+            plt.show()
+            im = prediction[0, 0, 5, :, :]
+            plt.imshow(im.cpu())
+            plt.title('generated image')
+            plt.show()
+        self.log("val_loss", loss)
+        return loss
+
+    def predict_step(self, batch, batch_idx):
+        image_dict, label = batch
+        prediction = self.forward(image_dict)
+        im = prediction[0, 0, 5, :, :].cpu()
+        plt.imshow(im)
+        plt.show()
+        return prediction
+
     def configure_optimizers(self):
         optimizer = torch.optim.Adam(self.parameters(), lr=1e-3)
         scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=5, gamma=0.5)
         return [optimizer], [scheduler]
 
+
 if __name__ == "__main__":
```

Comment:
"main" are great testing but please don't leave them here when you are satisfied

Reply:
I am not sure if use autoencoder will help, when I am sure, I will make another file

Reply:
Still the code online should not be for testing phase. Please remove the if__main__ from it and keep it in a local file.